feat(styling): add support for Bootstrap 5

gcko opened the pull request:
- Fix border-radius styling for `.headerrow` columns
- Ensure ui-slider maintains correct left margin
- handle changes to `.input-group` styling within Bootstrap
- Handle `.input-group` inner content height
- Appropriately size pagination (previous version was cutting off the bottom 6 pixels)
- Vertically align the pagination icons
- Ignore .naverc file (used for managing multiple node versions)

This is done in conjunction with the addition of a demonstration of Bootstrap 5 support in the [Angular Bootstrap Demos](https://github.com/ghiscoding/angular-slickgrid-demos/pull/5) repository.

ghiscoding commented:
Thanks this is great to support BS5... just a quick question, I assume this has no impact on BS3, BS4 implementations... correct?

ghiscoding commented on src/app/modules/angular-slickgrid/styles/slick-bootstrap.scss:
```diff
@@ -436,4 +442,40 @@
+  height: 40px;
+
+  .slick-pagination-nav .pagination .page-item a[class*="icon-seek-"] {
+    line-height: 1rem;
```
since I use `px` everywhere I think it might be better to add these two (`height` and `line-height`) as SASS variables (in the `_variables.scss` file), there's a few variables starting with `$pagination-` and so if you could add them there.

gcko replied:
will do

gcko replied:
Addressed by https://github.com/ghiscoding/Angular-Slickgrid/pull/657/commits/731fc0214a376f3872b7181a7d19a99a6673e5f0

gcko commented:
> Thanks this is great to support BS5... just a quick question, I assume this has no impact on BS3, BS4 implementations... correct?

@ghiscoding There is no impact on BS3/BS4, including any changes to existing visual bugs in BS4:

For Example, the following is how the [BS4 theme](https://github.com/ghiscoding/angular-slickgrid-demos/tree/master/bootstrap4-demo-with-translate) looks:
![Screen Shot 2021-01-03 at 17 11 32](https://user-images.githubusercontent.com/521927/103475271-bb973d80-4de6-11eb-9105-f6eba84fc6e4.png)

And in the [BS5 theme](https://github.com/ghiscoding/angular-slickgrid-demos/pull/5):
![Screen Shot 2021-01-03 at 17 13 48](https://user-images.githubusercontent.com/521927/103475309-0d3fc800-4de7-11eb-9eaf-a5c89484b97f.png)

ghiscoding commented:
After reviewing your PR a bit further, I got final changes to ask before merging
1. can you move the Flatpickr style changes into the `slick-plugins.scss` file which holds most of the Flatpickr styling
2. can you also move the Pagination style changes into `slick-component.scss` which is where all the Pagination styling are

The rest is all good in the bootstrap file.

ghiscoding commented:
Thanks for the confirmation, I took another look at your PR and I got 2 final changes that I'd like before merging. 
1. can you move the Flatpickr style change into `slick-plugins.scss`, that is where most of the Flatpickr styling is
2. can you also move the Pagination `slick-component.scss` which holds the Pagination styling

The rest can stay in the slick-bootstrap file

gcko commented:
> After reviewing your PR a bit further, I got final changes to ask before merging
> 
> 1. can you move the Flatpickr style changes into the `slick-plugins.scss` file which holds most of the Flatpickr styling
> 2. can you also move the Pagination style changes into `slick-component.scss` which is where all the Pagination styling are
> 
> The rest is all good in the bootstrap file.

@ghiscoding Addressed your comments in https://github.com/ghiscoding/Angular-Slickgrid/pull/657/commits/378d25fb89e2c4df92d19288d668e74b9e77cd22, please take a look

ghiscoding commented on src/app/modules/angular-slickgrid/styles/slick-plugins.scss:
```diff
@@ -790,6 +790,21 @@ input.flatpickr.form-control {
+    .slick-headerrow-columns {
+      .slick-headerrow-column {
+        .flatpickr {
+          input {
```
@gcko 
Thanks that is more organized this way. 
However, I found a small bug while recreating your PR in my other lib, there's a small issue with the Flatpickr styling, it shouldn't remove the left radius on the DateRange Filter (as can be seen below). 
To resolve this issue, you can add the `compound-input` class to the `input.compound-input`. Could you please apply the change and verify if that is all good on your side? You can test with [Example 25 - Range Filters](https://ghiscoding.github.io/Angular-Slickgrid/#/range)

![image](https://user-images.githubusercontent.com/643976/103501632-c967b580-4e1c-11eb-98e8-6c7a094900d5.png)

gcko replied:
@ghiscoding Good catch, the original intent is to ensure that compound filters that have a `.input-group-prepend` associated with Flatpickr do not border-left radius.

By adding `.compound-input` to `input`, it regresses the styling in this case:
![Screen Shot 2021-01-04 at 12 53 48](https://user-images.githubusercontent.com/521927/103502679-f3f15700-4e8c-11eb-9b58-1a71600d1da1.png)

However, by adding a sibling selector to `.flatpickr`, it allows for the removal of border radius specifically when there is a prepended filter:
![Screen Shot 2021-01-04 at 12 55 02](https://user-images.githubusercontent.com/521927/103502718-1a16f700-4e8d-11eb-9301-76f495d344ff.png)


```scss
...
.input-group-prepend + .flatpickr {
  input {
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
  }
}
...
```

gcko replied:
Addressed in https://github.com/ghiscoding/Angular-Slickgrid/pull/657/commits/91631a6acab308dce7ccf71c67001062b9618b1f

ghiscoding replied:
That is why I asked you to verify, 2 pair of eyes are better than one 😉 
BTW, `.operator + .flatpickr` would have also work but I think the group prepend is probably better

gcko replied:
🤣  Let me know if there is anything else you see that needs work

ghiscoding commented:
merge, expect a week or two before a new release since I got a few PRs in the queue. 

Thanks a lot for the contribution. ⭐ 

Do you have a rough idea of when BS5 is to be released? 
I'm guessing end of Q1

gcko commented:
> merge, expect a week or two before a new release since I got a few PRs in the queue.
> 
> Thanks a lot for the contribution. ⭐
> 
> Do you have a rough idea of when BS5 is to be released?
> I'm guessing end of Q1

Thanks for your quick execution on this, it's much appreciated. BS5 is already released (check https://getbootstrap.com/), it currently is Beta-1. I expect that to become a stable release by end of Q1.

ghiscoding commented:
This is now fixed and released in the new version `2.25.0`, see [Release Note](https://github.com/ghiscoding/Angular-Slickgrid/releases/tag/v2.25.0).

I also merged your Angular-Slickgrid-Demos PR and bump the Angular-Slickgrid version and used Bootstrap `@next` until it's released. Thanks a lot for your contribution on this 😉 

Cheers ⭐

gcko commented:
No worries, happy to participate!

Best,
Jared M. Scott
Founder | Variable.
+1-562-302-8063
+886-966463236


From: Ghislain B. <notifications@github.com>
Date: Thursday, January 7, 2021 at 02:11
To: ghiscoding/Angular-Slickgrid <Angular-Slickgrid@noreply.github.com>
Cc: Jared Scott <jared.scott@variable.team>, Mention <mention@noreply.github.com>
Subject: Re: [ghiscoding/Angular-Slickgrid] feat(styling): add support for Bootstrap 5 (#657)

This is now fixed and released in the new version 2.25.0, see Release Note<https://github.com/ghiscoding/Angular-Slickgrid/releases/tag/v2.25.0>.

I also merged your Angular-Slickgrid-Demos PR and bump the Angular-Slickgrid version and used Bootstrap @next until it's released. Thanks a lot for your contribution on this 😉

Cheers ⭐

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/ghiscoding/Angular-Slickgrid/pull/657#issuecomment-755466798>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAD7NR2KBCZ2D2NDVD2HGMTSYSRTJANCNFSM4VRSVACA>.
